src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs

```diff
@@ -281,41 +281,320 @@ public async Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestP
 
             _logger.LogInformation($"Merging pull request '{pullRequestUrl}' succeeded!");
         }
+        
+        /// <summary>
+        ///     Get updates required by coherency constraints.
+        /// </summary>
+        /// <param name="dependencies">Current set of dependencies.</param>
+        /// <param name="remoteFactory">Remote factory for remote queries.</param>
+        /// <returns>Dependencies with updates.</returns>
+        public async Task<Dictionary<DependencyDetail, DependencyDetail>> GetRequiredCoherencyUpdatesAsync(
+            IEnumerable<DependencyDetail> dependencies,
+            IRemoteFactory remoteFactory)
+        {
+            Dictionary<DependencyDetail, DependencyDetail> toUpdate = new Dictionary<DependencyDetail, DependencyDetail>();
+            // Leaves of the coherent dependency trees.  Basically 
+            // this means that the coherent dependency is not
+            // pointed to by another dependency, or is pointed to by only
+            // pinned dependencies.
+            // Examples:
+            //  - A->B(pinned)->C->D(pinned)
+            //      - C
+            //  - A->B->C->D
+            //      - D
+            //  - A->B
+            //      - B
+            //  - A->B->C(pinned)->D
+            //      - D
+            //      - B
+            //  - A->B(pinned)->C(pinned)
+            //      - None
+            IEnumerable<DependencyDetail> leavesOfCoherencyTrees =
```

Comment:
This appears to be 3 "does one match this at all".  Can we refactor the "is important" into a method (maybe a local method), and then make it just `if (!leavesOfCoherencyTrees.Any(SomeFunctionThatExplainsWhatItsDoing))`